### ISSUE TITLE
Refresh token at GHE is not working when using GIthub app authentication

### ISSUE THREAD

**Issue by Hi-Fi** [summerwind/actions-runner-controller#149]
GHES with repo and organization level runners tested only with PAT, and apparently there're some issue when using Github App that should be checked (see below).

Original issue related to possibility to register runners to enterprise level at #150

**Comment by ZacharyBenamram** [comment]
@Hi-Fi 

Seeing this when applying the GITHUB_ENTERPRISE_URL during autoscale

```
DEBUG	controller-runtime.manager.events	

Normal	{"object": {"kind":"HorizontalRunnerAutoscaler","namespace":"actions-runner-system","name":"runner-deployment-autoscaler","uid":"<>","apiVersion":"actions.summerwind.dev/v1alpha1","resourceVersion":"<>"}, 

"reason": "RunnerAutoscalingFailure", "message": "Get https://<GITHUB_ENTERPRISE_URL>/api/v3/repos/<ORG>/<REPO>/actions/runs: could not refresh installation id's <InstallationID>'s token: request &{Method:POST URL:https://api.github.com/app/installations/<InstallationID>/access_tokens
```

It looks as if the POST URL to refresh token is still using the public github url

**Comment by ZacharyBenamram** [comment]
I think the issue resides here: https://github.com/bradleyfalzon/ghinstallation/blob/81f7095e75089d688962bb75ba9814748d90f004/appsTransport.go#L57

**Comment by Hi-Fi** [comment]
@ZacharyBenamram Thank you for informing that issue, I only checked that I can register runners to GHES and that worked fine with changes in #146. Interesting how it can fail as whole client is created either for public or Enterprise here: https://github.com/summerwind/actions-runner-controller/blob/master/github/github.go#L55-L62. 

Of course I tried only with PATs, not with app, but still.

This issue was actually about possibility to register Enterprise level runners (in addition to current repository and organization level), but I create new issue related to that as there're more info related to the app issue.

**Comment by ZacharyBenamram** [comment]
I think this may fix: https://github.com/summerwind/actions-runner-controller/pull/151

**Comment by Hi-Fi** [comment]
@ZacharyBenamram Can you make image from #152 and test that? There's a bit more logic to get URL in correct form.

**Comment by ZacharyBenamram** [comment]
I think those api's can be accessed through the main route - (e.g. https://enterpriseUrl/app/installations/%v/access_tokens), but the change in https://github.com/summerwind/actions-runner-controller/pull/152 lgtm contingent on comments addressed

**Comment by Hi-Fi** [comment]
@ZacharyBenamram I think at GHE api always needs that /api/v3 -path, at least that's how the client library works. The URL building is directly from there, as it was included in creation function and not separate one.

**Comment by ZacharyBenamram** [comment]
I think its fine the way you wrote it - but fyi via https://docs.github.com/en/free-pro-team@latest/rest/reference/apps#create-an-installation-access-token-for-an-app looks like its available in the unversioned route
